network/dell/dnos10_config.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION = """
+---
+module: dnos10_config
+version_added: "2.2"
+author: "Senthil Kumar Ganesan (@skg_net)"
+short_description: Manage Dell OS10 configuration sections
+description:
+  - Dell OS10 configurations use a simple block indent file sytanx
+    for segementing configuration into sections.  This module provides
+    an implementation for working with Dell OS10 configuration sections in
+    a deterministic way.
+extends_documentation_fragment: dnos10
+options:
+  lines:
+    description:
+      - The ordered set of commands that should be configured in the
+        section.  The commands must be the exact same commands as found
+        in the device running-config.  Be sure to note the configuration
+        command syntax as some commands are automatically modified by the
+        device config parser.
+    required: false
+    default: null
+    aliases: ['commands']
+  parents:
+    description:
+      - The ordered set of parents that uniquely identify the section
+        the commands should be checked against.  If the parents argument
+        is omitted, the commands are checked against the set of top
```

Comment:
What decides if the commands are checked against "top level" or "global commands"?